src/widget/wratedisplay.cpp

```diff
@@ -0,0 +1,60 @@
+#include <QDebug>
+
+#include "widget/wratedisplay.h"
+
+#include "control/controlobject.h"
+#include "control/controlproxy.h"
+#include "util/math.h"
+
+WRateDisplay::WRateDisplay(const QString& group, QWidget* parent)
+        : WNumber(parent) {
+    m_pRateRangeControl = new ControlProxy(group, "rateRange", this, ControlFlag::NoAssertIfMissing);
+    m_pRateRangeControl->connectValueChanged(this, &WRateDisplay::setValue);
+    m_pRateDirControl = new ControlProxy(group, "rate_dir", this, ControlFlag::NoAssertIfMissing);
+    m_pRateDirControl->connectValueChanged(this, &WRateDisplay::setValue);
+}
+
+void WRateDisplay::setup(const QDomNode& node, const SkinContext& context) {
+    WNumber::setup(node, context);
+
+    QDomElement rateDisplayPosition = context.selectElement(node, "Position");
+    WRateDisplay::nodePosition = rateDisplayPosition.text() == "Top" ? VerticalPosition::Top : VerticalPosition::Bottom;
+
+    // Initialize the widget (overrides the base class initial value.
+    setValue();
+}
+
+void WRateDisplay::setValue() {
+    double range = m_pRateRangeControl->get();
+    double direction = m_pRateDirControl->get();
+
+    if(nodePosition == VerticalPosition::Top) {
+        qDebug() << "Position: TOP";
+    }
+    if(nodePosition == VerticalPosition::Bottom) {
+        qDebug() << "Position: BOTTOM";
+    }
+    if(direction > 0) {
+        qDebug() << "Direction > 0";
+    }
+    if(direction < 0) {
+        qDebug() << "Direction < 0";
+    }
+
+    QString sign('-');
+    if(nodePosition == VerticalPosition::Top && direction > 0) {
+        sign = '+';
+    }
+
+    if(nodePosition == VerticalPosition::Bottom) {
+        sign = '-';
+        if(direction < 0) {
+            sign = '+';
+        }
+    }
```

Comment:
`sign` is already initialized to `-` above, so this can be simplified:
``suggestion
    if (nodePosition == VerticalPosition::Bottom && direction < 0) {
        sign = '+';
    }
```